editor/src/clj/editor/atlas.clj

```diff
@@ -760,19 +776,31 @@
   (output uv-transforms    g/Any               (g/fnk [layout-data] (:uv-transforms layout-data)))
   (output layout-rects     g/Any               (g/fnk [layout-data] (:rects layout-data)))
 
-  (output texture-page-count g/Int             (g/fnk [layout-data max-page-size]
-                                                 (if (every? pos? max-page-size)
-                                                   (count (.layouts ^TextureSetGenerator$LayoutResult (:layout layout-data)))
-                                                   texture/non-paged-page-count)))
+  (output texture-page-count g/Int (g/fnk [_node-id layout-data max-page-size]
+                                     (let [page-count (calculate-texture-page-count layout-data max-page-size)]
+                                       (or (validation/prop-error :fatal _node-id
+                                                                  :validate-texture-page-count
+                                                                  texture-page-count-error-message
+                                                                  page-count)
+                                           page-count))))
 
   (output packed-page-images-generator g/Any   produce-packed-page-images-generator)
   (output texture-set-pb   g/Any               :cached produce-atlas-texture-set-pb)
 
   (output gpu-texture g/Any :cached
           (g/fnk [_node-id packed-page-images-generator texture-profile]
-            (-> (texture-util/construct-gpu-texture _node-id packed-page-images-generator texture-profile)
-                (texture/set-params {:min-filter gl/nearest
-                                     :mag-filter gl/nearest}))))
+            ;; NOTE: Temporary fix until we can properly disconnect the image nodes that invalidate this
+            (if (and (= (:sha1 packed-page-images-generator) (g/user-data _node-id :gpu-texture-sha1))
+                     (g/user-data _node-id :gpu-texture-cached))
+              (when-let [texture-ref (g/user-data _node-id :gpu-texture-cached)]
+                (.get texture-ref))
```

Comment:
I think this will return `nil` if the `gpu-texture` referenced by the `WeakReference` has been garbage-collected? We should treat this the same as a sha1 mismatch, right?